MdePkg/Include/Library/DebugLib.h

```diff
@@ -4,10 +4,24 @@
   The Debug library supports debug print and asserts based on a combination of macros and code.
   The debug library can be turned on and off so that the debug code does not increase the size of an image.
 
-  Note that a reserved macro named MDEPKG_NDEBUG is introduced for the intention
+  Note that a reserved macro named MDEPKG_NDEBUG is introduced with the intention
   of size reduction when compiler optimization is disabled. If MDEPKG_NDEBUG is
   defined, then debug and assert related macros wrapped by it are the NULL implementations.
 
+  Current NULL implementations of these macros rely on the fact that directly unreachable
+  code is pruned even without compiler optimization (which has been confirmed by generated
+  code size tests on supported compilers). The advantage of NULL implementations which consume
+  their arguments within directly unreachable code is that compilers understand this, and stop
+  warning about variables which would become unused when MDEPKG_NDEBUG is defined if the macros
+  had completely empty definitions. This approach allows additional compiler warnings to be
+  enabled and/or removes the need for MDEPKG_NDEBUG dependent code in C files purely to avoid
+  such warnings.
+
+  Note that macros here and elsewhere in EDK II use the standard approach of adding

```

Comment:
This note to me goes beyond what's needed for documentation purposes. Can it move to the commit message/PR description, or be dropped completely?

Reply:
Dropped.

Reply:
@leiflindholm (cc @mdkinney) - I think you probably wanted all the lines shown to be removed, not just the later one, I have now down this. I think the first part, end of earlier paragraph

> This approach allows additional compiler warnings to be enabled and removes the need for MDEPKG_NDEBUG dependent code in C files purely to avoid such warnings.

is already covered in the commit message.

The second part, whole of later paragraph:

> Note that macros here and elsewhere in EDK II use the standard approach of adding `do { ... } while (FALSE)` wrapping to ensure that they behave correctly with surrounding code (e.g. requiring a following ';' and not combining in unexpected ways with nearby conditionals).

I have now moved to the commit message (rather than just removing it), to try to address the concerns of both reviewers. Is this about right now?

Reply:
Ping @leiflindholm - following making additional wording changes requested by mdkinney, I'm genuinely unsure what you meant to capture with this comment. The last sentence of the upper paragraph, to me seems to complete the logic of the whole paragraph (i.e. why do we care that the compiler stops warning? is this good, or bad?).

The 'note' (which starts on the last of the four lines you highlighted), I guess you definitely did want removed. (And I have done so, moved to commit message.)

But could you clarify about the first part? Could I add those lines back?